catalog/nsili/catalog-nsili-transformer/src/main/java/com/connexta/alliance/nsili/transformer/DAGConverter.java

```diff
@@ -1200,6 +1244,27 @@ public static String getValueString(Collection<Serializable> collection) {
                 .collect(Collectors.joining(", "));
     }
 
+    private byte[] getThumbnail(String thumbnailUrlStr) {
+        byte[] thumbnail = null;
+
+        try {
+            URI thumbnailURI = new URI(thumbnailUrlStr);
+            ResourceResponse resourceResponse = null;
+            try {
+                resourceResponse = resourceReader.retrieveResource(thumbnailURI, new HashMap<>());
```

Comment:
Since this is being used in multiple threads and is using class variables to keep track of the state of parsing the Related File portion in order to find the THUMBNAIL, I create a new DAG Converter for each. The converter constructor is fairly lightweight so I thought this was a better approach than trying to create a custom state object to pass in and return from the addNsilRelatedFile method.